test/distributed/fsdp/test_fsdp_grad_acc.py

```diff
@@ -0,0 +1,146 @@
+# Owner(s): ["oncall: distributed"]
+
+import itertools
+import sys
+from unittest.mock import patch
+from typing import List, Tuple
+
+import torch
+from torch import distributed as dist
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import FSDPTest
+from torch.testing._internal.common_utils import (
+    TEST_WITH_DEV_DBG_ASAN,
+    instantiate_parametrized_tests,
+    parametrize,
+    run_tests,
+)
+
+if not dist.is_available():
+    print("Distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print(
+        "Skip dev-asan as torch + multiprocessing spawn have known issues",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+
+
+class TestGradAcc(FSDPTest):
```

Comment:
Can we test with CPU offload, backward prefetch, etc as well just to make sure things are working as expected?

Reply:
Testing with those orthogonal features makes sense to me! (though I may need some time to familiarize myself)